ReactCommon/react/renderer/mapbuffer/MapBufferBuilder.cpp

```diff
@@ -176,7 +176,7 @@ MapBuffer MapBufferBuilder::build() {
   // Create buffer: [header] + [key, values] + [dynamic data]
   int32_t bufferSize = keyValuesOffset_ + dynamicDataOffset_;
 
-  _header.bufferSize = bufferSize;
+  _header.bufferSize = static_cast<uint16_t>(bufferSize);
```

Comment:
I came across this and I'm not certain truncating to 16 bit is correct behavior. There was a recent change to move more of this class to use int32_t though.

Reply:
I'm going to test removing this + the case below locally in RNW. Though the above still makes sense (since string length is size_t).

Reply:
@NickGerleman What do you suggest for unblocking RNW?

Reply:
This indeed now builds without these 2 casts - removing.
Still, this gets implicitly cast to a narrower data type - OOC why do you think explicit casting is wrong here?